packages/h5wasm/src/models.ts

```diff
@@ -1,22 +1,5 @@
-import type {
-  CompoundTypeMetadata,
-  EnumTypeMetadata,
-  Group as H5WasmGroup,
-  Metadata,
-} from 'h5wasm';
+import type { Group as H5WasmGroup } from 'h5wasm';
 
 export type H5WasmEntity = ReturnType<H5WasmGroup['get']>;
 
 export type H5WasmAttributes = H5WasmGroup['attrs'];
-
-export interface CompoundMetadata extends Metadata {
-  compound_type: CompoundTypeMetadata;
```

Comment:
I now simply use `assertDefined` to make sure that `compound_type` exists when the HDF5 type `class` is `H5T_COMPOUND`.

Reply:
I guess we could we keep it and add the field `type: H5TClass.Compound` to avoid resorting to a type guard.

Reply:
H5Wasm's `Metadata` interface has all the fields and some of them (like `compound_type`) are optional. To avoid `assertDefined`, I would have to turn this `Metadata` type into a discriminated union (`CompoundMetadata | EnumMetadata | ...`), which is what I'm planning to do for h5grove. In h5wasm, however, there's typically one field that is different for each type (`compound_type`, `enum_type`, etc.), so `assertDefined` is much simpler I think.